src/Latte/Runtime/Filters.php

```diff
@@ -316,26 +316,33 @@ public static function strip(FilterInfo $info, string $s): string
 	 * @param  bool  $strip  stripping mode
 	 * @return string HTML
 	 */
-	public static function spacelessHtml(string $s, int $phase = null, bool &$strip = true): string
+	public static function spacelessHtml(string $s, ?int $phase = null, bool &$strip = true): string
 	{
 		if ($phase & PHP_OUTPUT_HANDLER_START) {
 			$s = ltrim($s);
 		}
 		if ($phase & PHP_OUTPUT_HANDLER_FINAL) {
 			$s = rtrim($s);
 		}
-		return preg_replace_callback(
+		$return = (string) preg_replace_callback( // Other cases
 			'#[ \t\r\n]+|<(/)?(textarea|pre|script)(?=\W)#si',
-			function ($m) use (&$strip) {
+			static function (array $m) use (&$strip): string {
 				if (empty($m[2])) {
-					return $strip ? ' ' : $m[0];
-				} else {
-					$strip = !empty($m[1]);
-					return $m[0];
+					return (string) $strip ? ' ' : $m[0];
 				}
+				$strip = !empty($m[1]);
+				return (string) $m[0];
 			},
 			$s
 		);
+		$return = (string) preg_replace_callback( // <script> for include JS file
+			'/<script\s*([^>]+?)>(?:\s*)<\/script>/',
```

Comment:
Will this regexp work with input like `<script title="> <\/script>">` ?

Reply:
This input is not supported now. I'll try to come up with a better solution.